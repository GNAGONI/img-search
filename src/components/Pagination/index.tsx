import * as React from 'react';
import Pagination from '@mui/material/Pagination';
import Stack from '@mui/material/Stack';

type Props = {
  page: number;
  totalPages: number;
  onChange: (event: React.ChangeEvent<unknown>, value: number) => void
};

const CustomPagination = (props: Props) => (
  <Stack spacing={2}>
    <Pagination
      count={props.totalPages}
      color="primary"
      page={props.page}
      onChange={props.onChange}
    />
  </Stack>
);


export default CustomPagination;
