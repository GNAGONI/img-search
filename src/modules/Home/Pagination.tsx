import * as React from 'react';
import { useSelector } from 'react-redux';
import { styled } from '@mui/material/styles';
import { getHomeImagesPage, getHomeImagesTotalPages } from './state/selectors';
import { changeImagePage } from './state/slice';
import CustomPagination from '../../components/Pagination';
import useAppDispatch from '../../hooks/useAppDispatch';

const PaginationWrapper = styled('div')({
  padding: '16px',
});

const Pagination = () => {
  const dispatch = useAppDispatch();
  const page = useSelector(getHomeImagesPage);
  const totalPages = useSelector(getHomeImagesTotalPages);
  const handlePaginationChange = (event: React.ChangeEvent<unknown>, value: number) => {
    dispatch(changeImagePage({ page: value }));
  };

  return (
    <PaginationWrapper>
      <CustomPagination
        totalPages={totalPages}
        page={page}
        onChange={handlePaginationChange}
      />
    </PaginationWrapper>
  );
};

export default Pagination;
