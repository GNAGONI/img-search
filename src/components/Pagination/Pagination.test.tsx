import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Pagination from './index';

describe('Pagination', () => {
  it('should have snapshot', () => {
    const mockCallback = jest.fn();
    const tree = renderer
      .create(
        <Pagination page={1} totalPages={10} onChange={mockCallback} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave correctly', () => {
    const mockCallback = jest.fn();
    render(<Pagination page={1} totalPages={3} onChange={mockCallback} />);

    const navigation = screen.getByRole('navigation', { name: 'pagination navigation' });
    expect(navigation).toBeInTheDocument();
    expect(navigation).toBeVisible();

    const list = screen.getByRole('list');
    expect(list).toBeInTheDocument();
    expect(list).toBeVisible();

    const buttonPrevious = screen.getByRole('button', { name: 'Go to previous page' });
    expect(buttonPrevious).toBeInTheDocument();
    expect(buttonPrevious).toBeVisible();

    const button1 = screen.getByRole('button', { name: 'page 1' });
    expect(button1).toBeInTheDocument();
    expect(button1).toBeVisible();
    expect(button1).toHaveAttribute("aria-current", "true");

    const button2 = screen.getByRole('button', { name: 'Go to page 2' });
    expect(button2).toBeInTheDocument();
    expect(button2).toBeVisible();

    const button3 = screen.getByRole('button', { name: 'Go to page 3' });
    expect(button3).toBeInTheDocument();
    expect(button3).toBeVisible();

    const buttonNext = screen.getByRole('button', { name: 'Go to next page' });
    expect(buttonNext).toBeInTheDocument();
    expect(buttonNext).toBeVisible();
  });
});

