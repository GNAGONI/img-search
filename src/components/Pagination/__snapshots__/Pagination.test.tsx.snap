// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should have snapshot 1`] = `
<div
  className="css-1p5q5e5-MuiStack-root"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          2
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          3
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          4
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          5
        </button>
      </li>
      <li>
        <div
          className="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 10"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          10
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
